src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1574,13 +1595,23 @@ public function getParameterBag(): ParameterBagInterface
             foreach ($this->buildParameters as $name => $value) {
                 $parameters[$name] = $value;
             }
-            $this->parameterBag = new FrozenParameterBag($parameters);
+            $this->parameterBag = new FrozenParameterBag($parameters, $this->deprecatedParameters ?? []);
         }
 
         return $this->parameterBag;
     }
 
 EOF;
+        if ($this->container->getParameterBag()->allDeprecated()) {
+            $code = preg_replace("/( +)__trigger_deprecation__/m", <<<'EOF'
+$1if (isset($this->deprecatedParameters[$name])) {
+$1    $deprecated = $this->deprecatedParameters[$name];
+$1    trigger_deprecation($deprecated['package'], $deprecated['version'], sprintf($deprecated['message'], $name));
```

Comment:
no need for the sprintf around, just pass the message and the name to trigger_deprecation

Reply:
Done (in ParameterBag as well).